pkg/sysinfo/sysinfo_linux.go

```diff
@@ -13,82 +13,83 @@ import (
+	}
+
+	return info
+}
 
-	if cgroupCpuMountpoint, err := cgroups.FindCgroupMountpoint("cpu"); err != nil {
+func checkCgroupCpu(quiet bool) *cgroupCpuInfo {
+	info := &cgroupCpuInfo{}
+	mountPoint, err := cgroups.FindCgroupMountpoint("cpu")
+	if err != nil {
 		if !quiet {
-			logrus.Warnf("%v", err)
-		}
-	} else {
-		_, err := ioutil.ReadFile(path.Join(cgroupCpuMountpoint, "cpu.cfs_period_us"))
-		sysInfo.CpuCfsPeriod = err == nil
-		if !sysInfo.CpuCfsPeriod && !quiet {
-			logrus.Warn("Your kernel does not support cgroup cfs period")
-		}
-		_, err = ioutil.ReadFile(path.Join(cgroupCpuMountpoint, "cpu.cfs_quota_us"))
-		sysInfo.CpuCfsQuota = err == nil
-		if !sysInfo.CpuCfsQuota && !quiet {
-			logrus.Warn("Your kernel does not support cgroup cfs quotas")
+			logrus.Warn(err)
 		}
+		return nil
 	}
 
-	// Checek if ipv4_forward is disabled.
-	if data, err := ioutil.ReadFile("/proc/sys/net/ipv4/ip_forward"); os.IsNotExist(err) {
-		sysInfo.IPv4ForwardingDisabled = true
-	} else {
-		if enabled, _ := strconv.Atoi(strings.TrimSpace(string(data))); enabled == 0 {
-			sysInfo.IPv4ForwardingDisabled = true
-		} else {
-			sysInfo.IPv4ForwardingDisabled = false
-		}
+	info.CpuCfsPeriod = cgroupEnabled(mountPoint, "cpu.cfs_period_us")
+	if !quiet && !info.CpuCfsPeriod {
+		logrus.Warn("Your kernel does not support cgroup cfs period")
 	}
 
-	// Check if bridge-nf-call-iptables is disabled.
-	if data, err := ioutil.ReadFile("/proc/sys/net/bridge/bridge-nf-call-iptables"); os.IsNotExist(err) {
-		sysInfo.BridgeNfCallIptablesDisabled = true
-	} else {
-		enabled, _ := strconv.Atoi(strings.TrimSpace(string(data)))
-		sysInfo.BridgeNfCallIptablesDisabled = enabled == 0
-	}
-	// Check if bridge-nf-call-ip6tables is disabled.
-	if data, err := ioutil.ReadFile("/proc/sys/net/bridge/bridge-nf-call-ip6tables"); os.IsNotExist(err) {
-		sysInfo.BridgeNfCallIp6tablesDisabled = true
-	} else {
-		enabled, _ := strconv.Atoi(strings.TrimSpace(string(data)))
-		sysInfo.BridgeNfCallIp6tablesDisabled = enabled == 0
+	info.CpuCfsQuota = cgroupEnabled(mountPoint, "cpu.cfs_quota_us")
+	if !quiet && !info.CpuCfsQuota {
+		logrus.Warn("Your kernel does not support cgroup cfs quotas")
 	}
+	return info
+}
 
-	// Check if AppArmor is supported.
-	if _, err := os.Stat("/sys/kernel/security/apparmor"); os.IsNotExist(err) {
-		sysInfo.AppArmor = false
-	} else {
-		sysInfo.AppArmor = true
-	}
+func cgroupEnabled(mountPoint, name string) bool {
+	_, err := ioutil.ReadFile(path.Join(mountPoint, name))
+	return err == nil
+}
 
-	// Check if Devices cgroup is mounted, it is hard requirement for container security.
-	if _, err := cgroups.FindCgroupMountpoint("devices"); err != nil {
-		logrus.Fatalf("Error mounting devices cgroup: %v", err)
+func readProcBool(path string) bool {
+	val, err := ioutil.ReadFile(path)
+	if err != nil {
+		return false
 	}
 
-	return sysInfo
+	enabled, _ := strconv.Atoi(strings.TrimSpace(string(val)))
```

Comment:
yeah buddy... I was focused on getting rid of `if{ ... } else { if {... } else {...}}` stuff, didn't even really start looking at these.

Will update.